huaweicloud/services/acceptance/gaussdb/data_source_huaweicloud_gaussdb_nosql_flavors_test.go

```diff
@@ -0,0 +1,147 @@
+					dc.CheckResourceExists(),
+					resource.TestCheckOutput("elem_engine_type", "cassandra"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "flavors.#"),
+				),
+			},
+			{
+				Config: testAccGaussDBNoSQLFlavors_cassandra(),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(dataSourceName, "engine_type", "cassandra"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "flavors.#"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccGaussDBNoSQLFlavors_mongodb(t *testing.T) {
+	dataSourceName := "data.huaweicloud_gaussdb_nosql_flavors.test"
+
+	dc := acceptance.InitDataSourceCheck(dataSourceName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      dc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGaussDBNoSQLFlavors_mongodb(),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(dataSourceName, "engine_type", "mongodb"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "flavors.#"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccGaussDBNoSQLFlavors_influxdb(t *testing.T) {
+	dataSourceName := "data.huaweicloud_gaussdb_nosql_flavors.test"
+
+	dc := acceptance.InitDataSourceCheck(dataSourceName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      dc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGaussDBNoSQLFlavors_influxdb(),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(dataSourceName, "engine_type", "influxdb"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "flavors.#"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccGaussDBNoSQLFlavors_redis(t *testing.T) {
+	dataSourceName := "data.huaweicloud_gaussdb_nosql_flavors.test"
+
+	dc := acceptance.InitDataSourceCheck(dataSourceName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      dc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGaussDBNoSQLFlavors_redis(),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(dataSourceName, "engine_type", "redis"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "flavors.#"),
+				),
+			},
+		},
+	})
+}
+
+func testAccGaussDBNoSQLFlavors_default() string {
+	return fmt.Sprintf(`
+data "huaweicloud_gaussdb_nosql_flavors" "test" {}
+
+output "elem_engine_type" {
+  value = element(tolist(data.huaweicloud_gaussdb_nosql_flavors.test.flavors[*].engine_type), 0)
```

Comment:
if just check first one , tolist is unnecessary

Reply:
sure